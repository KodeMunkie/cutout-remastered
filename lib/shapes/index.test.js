/* eslint-env jest */

const { randomShapeOf } = require('.');
const Rect = require('./Rect');

it('returns a shape', () => {
  const xBound = 10;
  const yBound = 10;
  const shape = randomShapeOf(['Rect'], xBound, yBound);

  expect(shape).toBeInstanceOf(Rect);
});

it('throws an error for invalid shape types', () => {
  const xBound = 10;
  const yBound = 10;

  expect(() => randomShapeOf(['Rhombus'], xBound, yBound)).toThrowErrorMatchingSnapshot();
});
