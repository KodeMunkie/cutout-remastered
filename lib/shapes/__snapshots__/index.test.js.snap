// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws an error for invalid shape types 1`] = `"rhombus is not a valid shape type"`;
